ui/client/components/graph/NodeDetailsContent.js

```diff
@@ -13,24 +13,34 @@ import * as JsonUtils from '../../common/JsonUtils';
 import Fields from "../Fields";
 import ParameterList from "./ParameterList";
 import ExpressionWithFixedValues from "./ExpressionWithFixedValues";
+import {v4 as uuid4} from "uuid";
+import MapVariable from "./NodeModal/MapVariable";
+import Variable from "./NodeModal/Variable";
 
 //move state to redux?
 // here `componentDidUpdate` is complicated to clear unsaved changes in modal
 export class NodeDetailsContent extends React.Component {
 
   constructor(props) {
     super(props);
+
     this.state = {
-      editedNode: props.node,
+      ...TestRenderUtils.stateForSelectTestResults(null, this.props.testResults),
+      editedNode: _.cloneDeep(props.node),
       codeCompletionEnabled: true,
-      testResultsToHide: new Set()
-    }
-    this.state = {
-      ...this.state,
-      ...TestRenderUtils.stateForSelectTestResults(null, this.props.testResults)
-    }
-    this.processNode = _.cloneDeep(props.node); //FIXME: props are modified without cloning @{ParameterList} breakks
-    this.nodeObjectDetails = ProcessUtils.findNodeObjectTypeDefinition(this.props.node, this.props.processDefinitionData.processDefinition)
+      testResultsToHide: new Set(),
+    };
+
+    this.generateUUID("fields", "service.parameters", "ref.parameters", "exceptionHandler.parameters");
```

Comment:
For `service.parameters`, `ref.parameters` and `exceptionHandler.parameters` it is redundant because in these cases names are unique - they will be changed if components would be changed in extension library